cmd/controlplane/api.go

```diff
@@ -36,18 +36,22 @@ func newAPICommand() *cobra.Command {
 				"commit":  version.GitCommit,
 			}).Info("Starting Kargo API Server")
 
+			cfg := config.ServerConfigFromEnv()
 			restCfg, err := kubernetes.GetRestConfig(ctx, os.GetEnv("KUBECONFIG", ""))
 			if err != nil {
 				return errors.Wrap(err, "error loading REST config")
 			}
 			kubeClient, err := kubernetes.NewClient(ctx, restCfg, kubernetes.ClientOptions{
+				KargoNamespace:    cfg.KargoNamespace,
 				NewInternalClient: newClientForAPI,
 			})
 			if err != nil {
-				return errors.Wrap(err, "error creating Kubernetes client")
+				return errors.Wrap(err, "create Kubernetes client")
+			}
+			internalClient, err := newClientForAPI(ctx, restCfg, kubeClient.Scheme())
```

Comment:
I think this can be improved. The way it is, we'll end up with _two_ distinct internal clients, and each of them will have their own in-memory cache. It won't be good for our memory consumption.

Can you possibly do something like this (pseudocode) so we have only one internal client?

```
internalClient, err := newClientForAPI(ctx, restCfg, kubeClient.Scheme())
// Handle error
kubeClient, err := kubernetes.NewClient(ctx, restCfg, kubernetes.ClientOptions{
  KargoNamespace:    cfg.KargoNamespace,
  NewInternalClient: func(...) ... {
    return internalClient
  }
})
```